src/Language/PureScript/AST/Exported.hs

```diff
@@ -144,3 +141,16 @@ isDctorExported ident (Just exps) ctor = test `any` exps
   test (TypeRef _ ident' Nothing) = ident == ident'
   test (TypeRef _ ident' (Just ctors)) = ident == ident' && ctor `elem` ctors
   test _ = False
+
```

Comment:
I'm not sure, this seems like premature optimisation to me. We would just as easily be able to handle doc comments in between the exports inside Sugar.Names by preserving the source spans there, right? And that would likely lead to better errors if we did it there.